tasmota/berry/drivers/Core200S.be

```diff
@@ -0,0 +1,712 @@
+                return k
+            end
+        end
+        return "invalid"
+    end
+
+    def toint()
+        return self.value
+    end
+
+    def ==(other)
+        return self.value == other.value
+    end
+
+end
+
+class OnOffState : StateEnum
+    static ON = OnOffState("on")
+    static OFF = OnOffState("off")
+    def init(value)
+        super(self).init({"off": 0, "on": 1}, value)
+    end
+end
+
+class FanMode : OnOffState
+    def init(value)
+        super(self).init(value)
+    end
+end
+
+class ChildLockMode : OnOffState
+    def init(value)
+        super(self).init(value)
+    end
+end
+
+class SleepMode : OnOffState
+    def init(value)
+        super(self).init(value)
+    end
+end
+
+class DisplayAutoOffMode : StateEnum
+    static OFF = DisplayAutoOffMode("off")
+    static ON = DisplayAutoOffMode("on")
+    def init(value)
+        super(self).init({"off": 1, "on": 0}, value)
+    end
+end
+
+class DisplayMode : StateEnum
+    def init(value)
+        super(self).init({"off": 0, "on": 100}, value)
+    end
+end
+
+class FanSpeedMode : StateEnum
+    static LOW = FanSpeedMode("low")
+    static MED = FanSpeedMode("med")
+    static FULL = FanSpeedMode("full")
+
+    def init(value)
+        super(self).init({"low": 1, "med": 2, "full": 3}, value)
+    end
+end
+
+class NightLightMode : StateEnum
+    static OFF = NightLightMode("off")
+    static LOW = NightLightMode("low")
+    static FULL = NightLightMode("full")
+    def init(value)
+        super(self).init({"off": 0, "low": 50, "full": 100}, value)
+    end
+end
+
+def modulo(num, mod)
+    return ((num % mod) + mod) % mod
+end
+
+def info_c2s(msg)
+    log("C2S: " + msg, 2)
+end
+
+def log_c2s(msg)
+    log("C2S: " + msg)
+end
+
+class Core200SSerial : serial
+    var msg_count
+    var read_buffer
+    var error_count
+    var expected_replies
+
+    def init()
+        super(self).init(16, 17, 115200)
+        self.reset_counters()
+        self.read_buffer = bytes("")
+    end
+
+    def deinit()
+        self.close()
+    end
+
+    static def calc_checksum(msg)
+        var checksum = 0xff
+        var i = 0
+        while i < msg.size()
+            if (i != 5)
+                checksum -= msg.get(i)
+                checksum = modulo(checksum, 256)
+            end
+            i += 1
+        end
+        return checksum
+    end
+
+    def send(msg)
+        self.send_nowait(msg)
+        tasmota.delay(100)
+        return self.read_msg()
+    end
+
+    def send_nowait(msg)
+        if msg.size() >= 2 && msg[1] == 0x22
+            self.expected_replies += 1
+        end
+
+        # increase and set message counter
+        self.msg_count += 1
+        msg[2] = self.msg_count
+
+        # insert checksum
+        msg[5] = self.calc_checksum(msg)
+
+        # send message
+        var i = 0
+        while i < msg.size()
+            self.write(msg[i])
+            i += 1
+        end
+
+        log_c2s("Send " + msg.tohex())
```

Comment:
You can consider also:
`log(f"Send {msg.tohex()}")`

Reply:
done